### DISCUSSION TITLE
Why not use code from git?

### DISCUSSION THREAD

**Discussion by GravisZro** [libgit2/libgit2#6783] [Q&A] [answered]
There are few code segments included that cause a lot of license complications that could easily be copied from the git source code which is also GPLv2. So the question is, why not utilize code from git which comes with multiple added benefits?

**Comment by arroz** [comment, marked as answer]
libgit2 has a linking exception, which allows you to do static linking without forcing the entire binary to become GPL (see beginning of https://github.com/libgit2/libgit2/blob/main/COPYING). My understanding is no "pure" GPLv2 code can be added otherwise this exception cannot be granted.

From what I see, all of the included 3rd party code has more permissive licenses than GPL (except winhttp?) so there should be no complications.

**Reply by ethomson** [reply]
What @arroz said is correct. But we also have different architectures.

Some git authors have allowed their code to be relicensed under the git license. Can you expand on the code that you're hoping for in libgit2 and the added benefits that you envision?

**Reply by GravisZro** [reply]
The biggest benefit is that you have another group of people reviewing the same code which means identifying/eliminating issues. However, there are lots of strange one-offs that make the licensing on this project is exceptionally complicated. The good news is that it seems much of the licensing complications are due to unnecessary library bundling.
